docs/fsharp/tour.md

```diff
@@ -1,4 +1,158 @@
+
+Because Discriminated Unions allow you to represent the recursive structure of the tree in the data type, operating on it is very straightforward and nets you correctness guarantees with pattern matching.  See below for more.
+
+## Pattern Matching
+
+[Pattern Matching](language-reference/pattern-matching.md) is the F# languag feature which enables correctness for operating on F# types.  In the above samples, you probably noticed quite a bit of `match x with ...` syntax.  This construct allows the compiler, which can understand the "shape" of data types, to enforce you to account for all possible cases when using a data type through what is known as Exhausting Pattern Matching.  This is incredibly powerful for correctness, and can be cleverly used to "lift" what would normally be a runtime concern into compile-time.
+
+[[!code-fsharp[PatternMatching](../../samples/snippets/fsharp/tour.fs#L496-L528)]
+
+You can also use the shorthand `function` construct for pattern matching, which is useful when you're writing functions which make use of [Partial Application](language-reference/functions/index.md#partial-application-of-arguments):
+
+[[!code-fsharp[PatternMatching](../../samples/snippets/fsharp/tour.fs#L530-L547)]
+
+Something you may have noticed is the use of the `_` pattern.  This is known as the "wildcard" pattern, which is basically a way of saying "I don't care what it is, since it doesn't matter to me".  Although convenient, you can accidentaly bypass Exhaustive Pattern Matching and no longer benefit from compile-time enforcements if you aren't careful.
+
+[Active Patterns](language-reference/active-patterns.md) are another powerful construct to use with pattern matching.  They allow you to partition input data into custom forms, decomposing them at the pattern match call site.  They can also be parameterized, thus allowing to define the partition as a function.  Expanding the previous example to support Active Patterns looks something like this:
+
+[[!code-fsharp[ActivePatterns](../../samples/snippets/fsharp/tour.fs#L530-L564)]
+
 >![NOTE]
-This is in-progress.
\ No newline at end of file
+The previous above example creates a `Result` type.  This has been added to the language with F# 4.1.  This document will be updated once F# 4.1 is shipped.
+
+## Optional Types
+
+One special case of Discriminated Union types is the Option Type, which is so useful that it's a part of the F# core library.
+
+[The Option Type](language-reference/options.md) is a type which represents one of two cases: a value, or nothing at all.  It is used in any scenario where a value may or may not result from a particular operation.  This then forces you to account for both cases, making it a compile-time concern rather than a runtime concern.  These are often used in APIs where `null` is used to represent "nothing" instead, thus eliminating the need to worry about `NulReferenceException` in many circumstances.
```

Comment:
The last sentence begs the question of value types. I'd assume option types would be analogous to nullables in C# and VB?